doc/modules/model_evaluation.rst

```diff
@@ -442,6 +449,45 @@ In the multilabel case with binary label indicators: ::
     for an example of accuracy score usage using permutations of
     the dataset.
 
+.. _top_k_accuracy_score:
+
+Top-k Accuracy score
+--------------------
+
+The :func:`top_k_accuracy_score` function is a generalization of
+:func:`accuracy_score`. The difference is that a prediction is considered
+correct as long as the true label is associated with one of the ``k`` highest
+predicted scores. :func:`accuracy_score` is the special case of `k = 1`.
+
+The function covers the multiclass classification case but not the binary and
+multilabel cases.
+
+If :math:`\hat{y}_{i,j}` is the predicted value of the :math:`i`-th sample for
+the :math:`j`-th largest predicted score and :math:`y_i` is the corresponding
+true value, then the fraction of correct predictions over
+:math:`n_\text{samples}` is defined as
+
+.. math::
+
+   \texttt{top-k accuracy}(y, \hat{y}) = \frac{1}{n_\text{samples}} \sum_{i=0}^{n_\text{samples}-1} \sum_{j=1}^{k} 1(\hat{y}_{i,j} = y_i)
```

Comment:
Strictly speaking j is in [0, k-1] but I guess it would mostly make the formula a bit ugly so we can keep it as is